packages/mds-policy-service/repository/index.ts

```diff
@@ -80,24 +84,35 @@ class PolicyReadWriteRepository extends ReadWriteRepository {
 
       if (statuses) {
         /** Turns statuses into expressions with params */
+        const PUBLISHED = "policy_json->>'publish_date' IS NOT NULL"
+        const NOT_PUBLISHED = "policy_json->>'publish_date' IS NULL"
+        const START_DATE_IN_PAST = "policy_json->>'start_date' IS NOT NULL AND policy_json->>'start_date' <= :now"
+        const START_DATE_IN_FUTURE = "policy_json->>'start_date' IS NOT NULL AND policy_json->>'start_date' > :now"
+        const END_DATE_IN_PAST = "policy_json->>'end_date' IS NOT NULL AND policy_json->>'end_date' <= :now"
+        const END_DATE_NULL_OR_IN_FUTURE = "(policy_json->>'end_date' IS NULL OR policy_json->>'end_date' > :now)"
+        const SUPERSEDED = 'superseded_by IS NOT NULL AND array_length(superseded_by, 1) >= 1'
+        const NOT_SUPERSEDED = 'superseded_by IS NULL'
+        const DELIMITER = ' AND '
         const statusToExpressionWithParams: {
           [key in Exclude<POLICY_STATUS, 'unknown'>]: { expression: string; params?: object }
         } = {
-          draft: { expression: "policy_json->>'publish_date' IS NULL" },
-          deactivated: { expression: 'superseded_by IS NOT NULL AND array_length(superseded_by, 1) >= 1' },
+          draft: {
+            expression: [NOT_PUBLISHED].join(DELIMITER)
+          },
+          deactivated: {
+            expression: [SUPERSEDED].join(DELIMITER)
+          },
```

Comment:
I could forsee a conflict where `SUPERSEDED` and `NOT_PUBLISHED` are both true for some things (if you write a new policy referencing an unpublished policy, if that's even possible, which maybe it shouldn't be)

Reply:
True. Who would be the source of truth on the policy status definitions? They can clarify whether "deactivated" can only be for policies that have been published. Also whether to add a check to the "supersede" action to only work for policies not in draft state (published).

Reply:
Probably @nelsonsantryl I'd imagine

Reply:
Hopefully I can clarify:

- A "deactivated" policy, must have been active at some point previously--> therefore the policy was published.
- The workflow to create a policy that supersedes another, is to go to the original **published** policy, and "clone" it (which will create the link back to the OG policy).  So yes you are correct.  There is no way to supersede a policy that is in a "draft" status.

Let me know if anything needs further clarification.